packages/flutter/lib/src/material/switch.dart

```diff
@@ -1462,7 +1530,7 @@ class _SwitchPainter extends ToggleablePainter {
       );
       final Paint outlinePaint = Paint()
         ..style = PaintingStyle.stroke
-        ..strokeWidth = 2
+        ..strokeWidth = trackOutlineWidth ?? 0.0
```

Comment:
strokeWidth = 0.0 isn't the same as "don't stroke the outline", see https://api.flutter.dev/flutter/dart-ui/Paint/strokeWidth.html

If we want trackOutlineWidth = null to mean don't draw the outline, then we could just skip drawing it here.

Reply:
Added a null check. Thanks!